packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,517 @@
+import 'package:flutter/material.dart';
+
+import '../trees.dart';
+import '../utils.dart';
+
+class Treemap extends StatefulWidget {
+  const Treemap({
+    @required this.rootNode,
+    @required this.levelsVisible,
+    @required this.width,
+    @required this.height,
+    @required this.onTap,
+  });
+
+  final TreemapNode rootNode;
+
+  /// The depth of children visible from this Treemap widget.
+  ///
+  /// For example, levelsVisible = 2 at root Treemap:
+  /// ```
+  /// _______________
+  /// |     Root    |
+  /// ---------------
+  /// |    l = 1    |
+  /// |  ---------  |
+  /// |  | l = 2 |  |
+  /// |  |       |  |
+  /// |  |       |  |
+  /// |  ---------  |
+  /// ---------------
+  /// ```
+  final int levelsVisible;
+
+  final double width;
+
+  final double height;
+
+  final VoidCallback onTap;
+
+  @override
+  _TreemapState createState() => _TreemapState();
+}
+
+enum PivotType { pivotByMiddle, pivotBySize }
```

Comment:
stylistically we should keep a StatefulWidget class directly above the corresponding State class. Move enum above Treemap or below the state class

Reply:
Done